### PR TITLE
feat: runTransaction support

### PR DESCRIPTION
**Pull request by asaarnak**
Add runTransaction support like here:
https://firebase.google.com/docs/firestore/solutions/aggregation#solution_cloud_functions

### CODE REVIEW

File: test/firestore_test.dart [hunk 1]
```diff
@@ -581,6 +581,49 @@ void main() {
...
+        await app.firestore().runTransaction((Transaction tx) async {
+          var doc2 = (await tx.get(doc2Ref)).data.getInt('value');
+          tx.update(doc1Ref, new UpdateData()..setInt('value', doc2),
+              Precondition(lastUpdateTime: doc1UpdateTime));
+          tx.delete(doc2Ref, Precondition(lastUpdateTime: doc2UpdateTime));
+        });
+        expect((await doc1Ref.get()).data.toMap(), {'value': 3});
+        expect((await doc2Ref.get()).exists, isFalse);
+      }, skip: '''
+        TODO https://github.com/pulyaevskiy/firebase-admin-interop/issues/19
```
Comment (asaarnak) [inline test/firestore_test.dart]: Created an issue https://github.com/pulyaevskiy/firebase-admin-interop/issues/19

Comment (asaarnak) [reply test/firestore_test.dart]: Yep, dart DateTime.parse() supports up to microsecond precision.
Firebase api says it's a nullable string, formatted as a UTC string.
https://firebase.google.com/docs/reference/functions/functions.firestore.DocumentSnapshot#updateTime

File: lib/src/firestore.dart [hunk 2]
```diff
@@ -884,6 +904,82 @@ class DocumentQuery {
...
+  }
+
+  /// Updates fields in the document referred to by the provided
+  /// [documentRef]. The update will fail if applied to a document that
+  /// does not exist.
+  /// Nested fields can be updated by providing dot-separated field path
+  /// strings.
+  /// update the document.
+  void update(DocumentReference documentRef, UpdateData data,
+      [js.Precondition precondition]) {
```
[the remark this answers is not in the document]
Comment (asaarnak) [reply lib/src/firestore.dart]: Done, good find!

File: lib/src/firestore.dart [hunk 3]
```diff
@@ -884,6 +904,82 @@ class DocumentQuery {
...
+    final nativeRef = documentRef.nativeInstance;
+    if (precondition != null) {
+      nativeInstance.update(nativeRef, docData, precondition);
+    } else {
+      nativeInstance.update(nativeRef, docData);
+    }
+  }
+
+  /// Deletes the document referred to by the provided [documentRef].
+  void delete(DocumentReference documentRef, [js.Precondition precondition]) {
```
[the remark this answers is not in the document]
Comment (asaarnak) [reply lib/src/firestore.dart]: Done.

File: test/firestore_test.dart (hunk 1 above)
Comment (asaarnak) [reply test/firestore_test.dart]: Couldn't find any good solution right now. Maybe this could be fixed in other PR with issue #19

**Comment by pulyaevskiy** [comment]
Published as 1.0.0-dev.15.0

**Pull request merged**